code/modules/mining/lavaland/loot/ashdragon_loot.dm

```diff
@@ -191,6 +192,14 @@
 	if(is_type_in_typecache(target, banned_turfs))
 		return
 
+	if(!is_mining_level(user.z)) //Will only spawn a few sparks if not on mining z level
+
+		if(world.time > timerMalfunction)
```

Comment:
You can just make use of the existing `timer` variable. That way you can just set that and have line 189 take care of the timing

Reply:
Makes sense, I wanted to use two variables to separate the behavior of malfunctioning from the other one, but I think it's way more simple the way you say.